rpcs3/Emu/RSX/rsx_decode.h

```diff
@@ -3369,8 +3369,8 @@ struct registers_decoder<NV4097_SET_INDEX_ARRAY_DMA>
 
 		index_array_type type() const
 		{
-			// Why truncate??
-			return to_index_array_type(static_cast<u8>(type_raw()));
+			// Must be a valid value
+			return static_cast<index_array_type>(static_cast<u8>(type_raw()));
```

Comment:
This second static cast is not needed. Just cast to index_array_type.

Reply:
It's needed, otherwise it's an UB.

Reply:
But index_array_type only ever interacts with this method, it can be changed to derive from u32.

Reply:
Maybe it should then. That's just how casting to enum works in general, it needs care.

Reply:
the truncation was not needed in the first place, it was a typo on it's type.